src/DataService/Account/Account.php

```diff
@@ -0,0 +1,499 @@
+    {
+        return $this->title;
+    }
+
+    /**
+     * Set The name of this object.
+     *
+     * @param string
+     */
+    public function setTitle($title)
+    {
+        if (strlen($title) > 64) {
+            throw new \InvalidArgumentException('Titles must not be longer than 64 characters.');
+        }
+        if (false !== strpos($title, '|')) {
+            throw new \InvalidArgumentException('Titles must not contain pipe (|) characters.');
+        }
+
+        $this->title = $title;
+    }
+
+    /**
+     * Returns The date and time this object was last modified.
+     *
+     * @return \DateTime
+     */
+    public function getUpdated(): \DateTime
+    {
+        return $this->updated;
+    }
+
+    /**
+     * Set The date and time this object was last modified.
+     *
+     * @param \DateTime
+     */
+    public function setUpdated($updated)
+    {
+        $this->updated = $updated;
+    }
+
+    /**
+     * Returns The id of the user that last modified this object.
+     *
+     * @return \Psr\Http\Message\UriInterface
+     */
+    public function getUpdatedByUserId(): UriInterface
+    {
+        return $this->updatedByUserId;
+    }
+
+    /**
+     * Set The id of the user that last modified this object.
+     *
+     * @param \Psr\Http\Message\UriInterface
+     */
+    public function setUpdatedByUserId($updatedByUserId)
+    {
+        $this->updatedByUserId = $updatedByUserId;
+    }
+
+    /**
+     * Returns This object's modification version.
+     *
+     * @return int
+     */
+    public function getVersion(): int
+    {
+        return $this->version;
+    }
+
+    /**
+     * Set This object's modification version.
+     *
+     * @param int
+     */
+    public function setVersion($version)
+    {
+        $this->version = $version;
+    }
+
+    public static function loadAllAccounts(UserSession $userSession)
+    {
+        // @todo Implement global paging on search results.
+        // @todo Implement global handling of adding schema.
+        // @todo Implement global handling of adding cjson (Request object?)
+        $promise = $userSession->requestAsync('GET', static::ACCOUNT_URI, [
+            'query' => [
+                'schema' => '1.0',
+                'form' => 'cjson',
+            ],
+        ])->then(function (ResponseInterface $response) {
+            $encoders = [new JsonEncoder()];
+            $normalizers = [new ObjectNormalizer()];
+
+            $serializer = new Serializer($normalizers, $encoders);
+            $results = \GuzzleHttp\json_decode($response->getBody(), true);
+            $accounts = [];
+            foreach ($results['entries'] as $entry) {
+                // @todo Implement a QueryResult object that is returned, that in turn deserializes each object?
+                $encoded = \GuzzleHttp\json_encode($entry);
+                /** @var self $account */
+                $account = $serializer->deserialize($encoded, self::class, 'json', [
```

Comment:
I'm guessing this deserialize() method is responsible for mapping the response entries as properties of this Account class, is that right?

Is there some documentation link about what The Platform should be returning? I haven't found the Account object there.

Reply:
That's right, it's [the ObjectNormalizer](https://symfony.com/doc/3.4/components/serializer.html) above that looks for public properties and set() methods, calling them automatically.

The account object is at https://docs.theplatform.com/help/wsf-account-endpoint

Reply:
It would be nice to add that link as @see before the class definition

I've seen loadAllAccounts() has been removed here anyway: https://github.com/Lullabot/mpx-php/pull/43/files. It would had been better to review all these pull request starting from newer...